chapter5

rlatls1997 opened the pull request:
5장 웹 서버 리팩토링

rlatls1997 committed:
refactor : 5장 - 요청데이터 처리 클래스 분리

rlatls1997 committed:
refactor : 5장 - 응답데이터 처리 클래스 분리

rlatls1997 committed:
refactor : 5장 - 응답데이터 처리 로직 다형성 적용

seung-hun-h commented on src/main/java/user/UserController.java:
```diff
@@ -0,0 +1,121 @@
+		return password.equals(user.getPassword());
+	}
+
+	private String getUsers() {
+		Collection<User> users = DataBase.findAll();
+
+		StringBuilder sb = new StringBuilder();
+		sb.append("<html>");
+		sb.append("<head>");
+		sb.append("<title>사용자 목록</title>");
+		sb.append("</head>");
+		sb.append("<body>");
+
+		for (User user : users) {
+			sb.append("<div>사용자명:")
+				.append(user.getName())
+				.append(", 이메일:")
+				.append(user.getEmail())
+				.append("</div>");
+		}
+
+		sb.append("</body>");
+		sb.append("</html>");
+
+		return sb.toString();
+	}
```
메서드 이름이 getUsers인데 html 형식의 String이 반환되어 어색한거 같습니다

seung-hun-h commented on src/main/java/user/UserController.java:
```diff
@@ -0,0 +1,121 @@
+			return;
+		}
+
+		throw new IllegalStateException("Request path not found. path:" + path);
```
IllegalStateException이 맞나라는 생각이 드네요. UnSupportedOperationException 같은게 문맥에 맞지 않을까요?
저도 확실하진 않네요

seung-hun-h reviewed (approved):
리뷰 확인하고 머지하면 될 것 같네요

rlatls1997 committed:
refactor : 5장 - Controller코드 리팩터링
- exception 타입 변경(IllegalStateException -> UnsupportedOperationException)
- 메서드 네임 수정(getUsers -> getUserListHtml
